### PR TITLE
docs - add content for column-level privs

### PR DESCRIPTION
**Pull request by lisakowen**
updates in this PR:
- pg_attribute sys catalog ref page - add attacl
    - additional - also add missing attcollation, attoptions, attfdwoptions
    - misc formatting edits
- INSERT, SELECT, UPDATE ref pages - minor content updates describing the privileges required
- GRANT ref page - add column variant, doc edits/updates to make equivalent with postgresql 9.3 ref page
    - additional - add GRANT ... ON ALL [TABLES/SEQUENCES/FUNCTIONS] IN SCHEMA variants and associated content
    - misc edits to "GRANT on Protocols" section; this config is not just for custom protocols
- REVOKE ref page - add column variant, add Notes from postgresql 9.3 ref page
    - additional - add REVOKE ... ON ALL [TABLES/SEQUENCES/FUNCTIONS IN SCHEMA] variants
- SQL 2008 Optional Feature Compliance page - update/insert/select column-level privileges - YES
- Summary of Greenplum Features - remove references to column-level update/insert/select not being supported
- "Managing Roles and Privileges" page - add "grant access to column" example, remove column-level privileges reference in discussion of simulating row and column level access, misc edits
- "Configuring Database Authorization" page - remove column-level privileges reference in discussion of simulating row and column level access

doc review site links:
- http://docs-lisa-gpdb6.cfapps.io/600/ref_guide/sql_commands/GRANT.html
- http://docs-lisa-gpdb6.cfapps.io/600/ref_guide/sql_commands/REVOKE.html
- http://docs-lisa-gpdb6.cfapps.io/600/ref_guide/sql_commands/INSERT.html
- http://docs-lisa-gpdb6.cfapps.io/600/ref_guide/sql_commands/SELECT.html
- http://docs-lisa-gpdb6.cfapps.io/600/ref_guide/sql_commands/UPDATE.html
- http://docs-lisa-gpdb6.cfapps.io/600/ref_guide/system_catalogs/pg_attribute.html#topic1

### CODE REVIEW

File: gpdb-doc/dita/admin_guide/roles_privs.xml [hunk 1]
```diff
@@ -392,15 +392,19 @@
...
         that database. It only grants all of the database-level privileges
         (<codeph>CONNECT</codeph>, <codeph>CREATE</codeph>, <codeph>TEMPORARY</codeph>) to the
         database itself.</note>
       <p>Use the <codeph>GRANT</codeph> SQL command to give a specified role privileges on an
-        object. For example:</p>
+        object. For example, to grant the role named <codeph>jsmith</codeph> insert
+        privileges on the table named <codeph>mytable</codeph>:</p>
       <p>
         <codeblock>=# GRANT INSERT ON mytable TO jsmith;</codeblock>
+        Similarly, to grant <codeph>jsmith</codeph> select access to only the column named
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/roles_privs.xml]: select access to only the column -> select privileges only to the column

File: gpdb-doc/dita/admin_guide/roles_privs.xml [hunk 2]
```diff
@@ -417,13 +421,13 @@
...
       <body>
-        <p>Row-level or column-level access is not supported, nor is labeled security. Row-level and
-          column-level access can be simulated using views to restrict the columns and/or rows that
-          are selected. Row-level labels can be simulated by adding an extra column to the table to
+        <p>Row-level access is not supported, nor is labeled security. You can
+          simulate row-level access by using views to restrict the rows that
+          are selected. You can simulate row-level labels by adding an extra column to the table to
           store sensitivity information, and then using views to control row-level access based on
-          this column. Roles can then be granted access to the views rather than the base table.
+          this column. You can then grant roles access to the views rather than the base table.
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/roles_privs.xml]: than the base -> than to the base

File: gpdb-doc/dita/admin_guide/roles_privs.xml [hunk 3]
```diff
@@ -417,13 +421,13 @@
       </p>
     </body>
     <topic id="topic7" xml:lang="en">
-      <title>Simulating Row and Column Level Access Control</title>
+      <title>Simulating Row Level Access Control</title>
       <body>
-        <p>Row-level or column-level access is not supported, nor is labeled security. Row-level and
-          column-level access can be simulated using views to restrict the columns and/or rows that
-          are selected. Row-level labels can be simulated by adding an extra column to the table to
+        <p>Row-level access is not supported, nor is labeled security. You can
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/roles_privs.xml]: Edit to something like: Greenplum does not support row-level access or row-level, labeled security.

File: gpdb-doc/dita/ref_guide/sql_commands/GRANT.xml [hunk 4]
```diff
@@ -3,11 +3,18 @@
...
     TO {<varname>rolename</varname> | PUBLIC} [, ...] [WITH GRANT OPTION]
 
+GRANT { { SELECT | INSERT | UPDATE | REFERENCES } ( <varname>columnname</varname> [, ...] )
+    [,...] | ALL [ PRIVILEGES ] ( <varname>columnname</varname> [, ...] ) }
+    ON [ TABLE ] <varname>tablename</varname> [, ...]
+    TO { <varname>rolename</varname> | PUBLIC } [, ...] [ WITH GRANT OPTION ]
+
 GRANT { {USAGE | SELECT | UPDATE} [,...] | ALL [PRIVILEGES] }
-    ON SEQUENCE <varname>sequencename</varname> [, ...]
+    ON } SEQUENCE <varname>sequencename</varname> [, ...]
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/GRANT.xml]: The first } here seems incorrect.  Seems like it should be {

File: gpdb-doc/dita/ref_guide/sql_commands/GRANT.xml [hunk 5]
```diff
@@ -3,11 +3,18 @@
...
+         | ALL SEQUENCES IN SCHEMA schemaname [, ...] }
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/GRANT.xml]: schemaname should be tagged as &lt;varname&gt;

File: gpdb-doc/dita/ref_guide/sql_commands/GRANT.xml [hunk 6]
```diff
@@ -24,8 +31,9 @@ GRANT { USAGE | ALL [ PRIVILEGES ] }
     TO { [ GROUP ] <varname>rolename</varname> | PUBLIC } [, ...] [ WITH GRANT OPTION ]
 
 GRANT { EXECUTE | ALL [PRIVILEGES] }
-    ON FUNCTION <varname>funcname</varname> ( [ [<varname>argmode</varname>] [<varname>argname</varname>] <varname>argtype</varname> [, ...] 
+    ON { FUNCTION <varname>funcname</varname> ( [ [<varname>argmode</varname>] [<varname>argname</varname>] <varname>argtype</varname> [, ...] 
 ] ) [, ...]
+        | ALL FUNCTIONS IN SCHEMA schema_name [, ...] }
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/GRANT.xml]: Same here (varname).  Also stick with either schemaname or schema_name, probably the former for consistency.

File: gpdb-doc/dita/ref_guide/sql_commands/GRANT.xml [hunk 7]
```diff
@@ -46,11 +54,17 @@ GRANT <varname>parent_role</varname> [, ...]
...
 membership in a role.</p><sectiondiv id="section4"><b>GRANT on Database Objects</b><p>This variant of the <codeph>GRANT</codeph> command gives specific privileges
 on a database object to one or more roles. These privileges are added
-to those already granted, if any. </p><p>The key word <codeph>PUBLIC</codeph> indicates that the privileges
+to those already granted, if any. </p>
+  <p>The <codeph>GRANT</codeph> command also includes an option to grant
+    privileges on all objects of the same type within one or more schemas.
+    While this functionality is currently supported only for tables, sequences,
+    and functions, note that <codeph>ALL TABLES</codeph> is considered to
+    include views and foreign tables.</p>
+  <p>The key word <codeph>PUBLIC</codeph> indicates that the privileges
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/GRANT.xml]: key word -> keyword

File: gpdb-doc/dita/ref_guide/sql_commands/GRANT.xml [hunk 8]
```diff
@@ -64,40 +78,48 @@ the role that created it), as the owner has all privileges by default.
...
-<codeph>EXECUTE</codeph> privilege for functions; and <codeph>USAGE</codeph>
-privilege for languages. The object owner may of course revoke these
-privileges.</p></sectiondiv><sectiondiv id="section5"><b>GRANT on Roles</b><p>This variant of the <codeph>GRANT</codeph> command grants membership
+options for the object, too.</p>
+  <p>Greenplum Database grants default privileges on some types of objects
+    to <codeph>PUBLIC</codeph>. No privileges are granted to <codeph>PUBLIC</codeph>
+    by default on tables, table columns, sequences, foreign data wrappers,
+    foreign servers, large objects, schemas, or tablespaces. For other types
+    of objects, the default privileges granted to <codeph>PUBLIC</codeph>
+    are as follows: <codeph>CONNECT</codeph> and <codeph>TEMPORARY</codeph>
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/GRANT.xml]: Break this into a bullet list?

File: gpdb-doc/dita/ref_guide/sql_commands/GRANT.xml [hunk 9]
```diff
@@ -122,12 +144,18 @@ met). Essentially this allows the grantee to look up objects within the
...
      </plentry><plentry><pt>ALL PRIVILEGES</pt><pd>Grant all of the available privileges at once. The <codeph>PRIVILEGES</codeph>
 key word is optional in Greenplum Database, though it is required by
 strict SQL. </pd></plentry><plentry><pt>PUBLIC</pt><pd>A special group-level role that denotes that the privileges are to
 be granted to all roles, including those that may be created later. </pd></plentry><plentry><pt>WITH GRANT OPTION</pt><pd>The recipient of the privilege may in turn grant it to others.</pd></plentry><plentry><pt>WITH ADMIN OPTION</pt><pd>The member of a role may in turn grant membership in the role to
-others.</pd></plentry></parml></section><section id="section8"><title>Notes</title><p>Database superusers can access all objects regardless of object privilege
+others.</pd></plentry></parml></section><section id="section8"><title>Notes</title>
+<p>A user may perform <codeph>SELECT</codeph>, <codeph>INSERT</codeph>,
+  etc. on a column if he holds that privilege for either the specific
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/GRANT.xml]: etc. -> and so forth

File: gpdb-doc/dita/ref_guide/sql_commands/GRANT.xml [hunk 10]
```diff
@@ -158,8 +186,15 @@ setting the privileges on more than one object per command. </p><p>Greenplum Dat
...
-        granted by an assumed <i>system</i> entity. </p><p>The SQL standard allows setting privileges for individual columns
-within a table.</p><p>The SQL standard provides for a <codeph>USAGE</codeph> privilege on
-other kinds of objects: character sets, collations, translations, domains.
-</p><p>Privileges on databases, tablespaces, schemas, and languages are Greenplum
+        granted by an assumed <i>system</i> entity. </p>
+     <p>According to the SQL standard, grant options can be granted to <codeph>PUBLIC</codeph>;
+       Greenplum Database supports granting grant options only to roles.</p>
+     <p>The SQL standard provides for a <codeph>USAGE</codeph> privilege on
+other kinds of objects: character sets, collations, translations.</p>
+     <p>In the SQL standard, sequences only have a <codeph>USAGE</codeph> privilege,
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/GRANT.xml]: privilege, -> privilege.

**Comment by lisakowen** [comment]
thanks for reviewing, @dyozie .  i made your requested edits.

**Comment by lisakowen** [comment]
looking for a technical review of this content, if anyone is interested...

**Comment by lisakowen** [comment]
@Eulerizeit @danielgustafsson @hlinnaka @melanieplageman

**Pull request merged**